codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/protocol/aws/SerializeMiddleware.java

```diff
@@ -0,0 +1,223 @@
+    
+                """,
+                MapUtils.of(
+                        "input", ctx.getSymbolProvider().toSymbol(input),
+                        "request", generator.getApplicationProtocol().getRequestType(),
+                        "route", handleHttp(),
+                        "serialize", handlePayload(),
+                        "errorf", GoStdlibTypes.Fmt.Errorf
+                ));
+    }
+
+    private GoWriter.Writable handleHttp() {
+        return goTemplate("""
+                    input, ok := in.Parameters.($input:P)
+                    if !ok {
+                        return out, metadata, $errorf:T("unexpected input type %T", in.Parameters)
+                    }
+                    _ = input
+                
+                    req, ok := in.Request.($request:P)
+                    if !ok {
+                        return out, metadata, $errorf:T("unexpected transport type %T", in.Request)
+                    }
+                
+                    req.Method = $methodPost:T
+                    req.URL.Path = "/service/$service:L/operation/$operation:L"
+                    req.Header.Set("smithy-protocol", $protocol:S)
+                
+                    $contentTypeHeader:W
+                    $acceptHeader:W
+                
+                """,
+        MapUtils.of(
+                "input", ctx.getSymbolProvider().toSymbol(input),
+                "request", generator.getApplicationProtocol().getRequestType(),
+                "methodPost", GoStdlibTypes.Net.Http.MethodPost,
+                "service", ctx.getService().getId().getName(),
+                "operation", operation.getId().getName(),
+                "protocol", generator.getProtocol(),
+                "contentTypeHeader", setContentTypeHeader(),
+                "acceptHeader", acceptHeader(),
+                "serialize", handlePayload(),
+                "errorf", GoStdlibTypes.Fmt.Errorf
+                ));
+    }
+
+    private GoWriter.Writable handlePayload() {
+        return goTemplate("""
+            
+                jsonEncoder := $encoder:T()
+            
+                err = $serialize:L(input, jsonEncoder.Value)
+                if err != nil {
+                    return out, metadata, &$error:T{Err: err}
+                }
+
+            
+                payload := $reader:T(jsonEncoder.Bytes())
+                if req, err = req.SetStream(payload); err != nil {
+                    return out, metadata, &$error:T{Err: err}
+                }
+
+
+
+                in.Request = req
+            """,
+    MapUtils.of(
+            "serialize", getSerializerName(input),
+            "encoder", SmithyGoTypes.Encoding.Json.NewEncoder,
+            "reader", GoStdlibTypes.Bytes.NewReader,
+            "error", SmithyGoTypes.Smithy.SerializationError
+    ));
+    }
+
+    private GoWriter.Writable setContentTypeHeader() {
+        if (input.hasTrait(BackfilledInputOutputTrait.class)) {
```

Comment:
This isn't necessary for awsjson10 - since we put a synthetic empty input struct on each operation that didn't have one, and you're generating the serializers for them and calling them, you're guaranteed to get an empty object `{}` as dictated by the protocol. https://smithy.io/2.0/aws/protocols/aws-json-1_0-protocol.html#empty-body-serialization